lib/stripe/error.ex

```diff
@@ -163,7 +163,7 @@ defmodule Stripe.Error do
           %{raw_error: error_data, http_status: status}
           |> maybe_put(:card_code, error_data |> Map.get("code") |> maybe_to_atom())
```

Comment:
```suggestion
          |> maybe_put(:card_code, error_data |> Map.get("code"))
```

We shouldn't be doing this casting, either. At least it is still more "controlled" on the Stripe side, but it still makes me worry about these types of casting from systems we do not control.

What do you think?

---

Also, based on https://github.com/stripe/stripe-go/blob/f333354373028fc1794afbc8c2e3dc3648fddfd5/error.go#L25C1-L236

It seems we are missing out values, they probably should be code-gen or removed altogether from the typespec and use string 🤷🏻

Reply:
Yeah I agree. All uses of `String.to_atom` should really be `String.to_exisiting_atom` here because of the security reasons mentioned in the issue. If that's not possible because of e.g. forward-compatibility with Stripe errors, we should just be using strings.

Reply:
Any strong reason to be using atoms, to begin with?

Pattern matching on a string is the same, so this is purely an aesthetic difference but with some concerns in the future.

Reply:
Only benefits for using atoms here is is union literal types which can aid in static typing and documentation.

Reply:
At what cost, thou, especially since we must maintain that list manually, for what I can tell.

I rather point to their documentation than deal with atoms and manually maintained lists.

Reply:
Yes agree, imo any atom values here should be strings.